jobbergate-core/jobbergate_core/auth/core.py

```diff
@@ -0,0 +1,316 @@
+"""
+Utilities for handling authentication in the Jobbergate system.
+"""
+import time
+from collections import namedtuple
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, cast
+
+import httpx
+from loguru import logger
+
+from .exceptions import AuthenticationError, TokenError
+from .token import Token, TokenType
+
+
+_LoginInformation = namedtuple(
+    "_LoginInformation",
+    ["verification_url", "wait_interval", "device_code"],
+)
+
+
+@dataclass
+class JobbergateAuth:
+    """
+    High-level class used to handle authentication in Jobbergate.
+
+    After an instance of this class is created, it can be used to authenticate requests
+    from the `requests`_ library or the `httpx`_ library by passing it to the ``auth``
+    parameter on the request (see examples bellow).
```

Comment:
```suggestion
    After an instance of this class is created, it can be used to authenticate requests
    from both `requests`_  and `httpx`_ packages by passing it to the ``auth``
    parameter on the request (see examples bellow).
```

Reply:
solved in [9698a75](https://github.com/omnivector-solutions/jobbergate/pull/228/commits/9698a75a2f5cfe2f7e8c24136486f3e5fb4e5bea)